packages/app/src/debug/DebugContainer.cy.tsx

```diff
@@ -0,0 +1,161 @@
+import { DebugSpecsFragmentDoc } from '../generated/graphql-test'
+import DebugContainer from './DebugContainer.vue'
+import { defaultMessages } from '@cy/i18n'
+import { useLoginConnectStore } from '@packages/frontend-shared/src/store/login-connect-store'
+import { specsList } from './utils/DebugMapping'
+import { CloudRunStubs } from '@packages/graphql/test/stubCloudTypes'
+
+describe('<DebugContainer />', () => {
+  context('empty states', () => {
+    const validateEmptyState = (expectedMessage: string) => {
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        render: (gqlVal) => {
+          return (
+            <DebugContainer
+              gql={gqlVal}
+            />
+          )
+        },
+      })
+
+      cy.findByTestId('debug-empty').contains(expectedMessage)
+    }
+
+    it('shows not logged in', () => {
+      validateEmptyState(defaultMessages.debugPage.notLoggedIn)
+    })
+
+    it('is logged in', () => {
+      const loginConnectStore = useLoginConnectStore()
+
+      loginConnectStore.setUserFlag('isLoggedIn', true)
+
+      validateEmptyState(defaultMessages.debugPage.notConnected)
+    })
+
+    it('has no runs', () => {
+      const loginConnectStore = useLoginConnectStore()
+
+      loginConnectStore.setUserFlag('isLoggedIn', true)
+      loginConnectStore.setProjectFlag('isProjectConnected', true)
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        render: (gqlVal) => {
+          return (
+            <DebugContainer
+              gql={gqlVal}
+            />
+          )
+        },
+      })
+
+      cy.findByTestId('debug-empty').contains(defaultMessages.debugPage.noRuns)
+    })
+  })
+
+  describe('render specs and tests', () => {
+    it('renders data when logged in and connected', () => {
+      const loginConnectStore = useLoginConnectStore()
+
+      loginConnectStore.setUserFlag('isLoggedIn', true)
+      loginConnectStore.setProjectFlag('isProjectConnected', true)
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        onResult: (result) => {
+          if (result.currentProject?.cloudProject?.__typename === 'CloudProject') {
+            const test = result.currentProject.cloudProject.runByNumber
+            const other = CloudRunStubs.failingWithTests as typeof test
+
+            result.currentProject.cloudProject.runByNumber = other
+          }
+        },
+        render: (gqlVal) => {
+          return (
+            <DebugContainer
+              gql={gqlVal}
+            />
+          )
+        },
+      })
+
+      // Only asserting that it is rendering the components for failed specs
+      cy.findByTestId('debug-header').should('be.visible')
+      cy.findByTestId('debug-spec-item').should('be.visible')
+    })
+  })
+
+  describe('testing util function: debugMapping', () => {
+    const mappingList = (specs: {id: string}[], tests: {specId: string, id: string}[]) => {
```

Comment:
This function looks like it returns a function with the exact same arguments; what is the purpose of having `mappingList`? Can we just call `specsList` instead?

Reply:
Yes!  Good catch. Will update.